### PR TITLE
Refactor ParMapTest from Kotest Plugin to Kotlin-test runtime #3191

### PR DESCRIPTION
**Pull request by asmadsen**

### CODE REVIEW

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 1]
```diff
@@ -17,10 +16,12 @@ import io.kotest.property.arbitrary.string
...
 import kotlin.time.Duration.Companion.milliseconds
 import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.test.runTest
 import kotlinx.coroutines.withTimeoutOrNull
+import kotlin.test.Test
 
-class ParMapTest : StringSpec({
-  "parMap is stack-safe" {
+class ParMapTest {
+  @Test fun parMapIsStackSafe() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:24]:
```suggestion
  @Test fun parMapIsStackSafe() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 2]
```diff
@@ -89,7 +90,7 @@ class ParMapTest : StringSpec({
     }
   }
 
-  "parMapOrAccumulate is stack-safe" {
+  @Test fun parMapOrAccumulateIsStackSafe() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:93]:
```suggestion
  @Test fun parMapOrAccumulateIsStackSafe() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 3]
```diff
@@ -144,15 +145,15 @@ class ParMapTest : StringSpec({
     } shouldBe null
   }
 
-  "parMapOrAccumulate accumulates shifts" {
+  @Test fun parMapOrAccumulateAccumulatesShifts() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:148]:
```suggestion
  @Test fun parMapOrAccumulateAccumulatesShifts() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 4]
```diff
@@ -144,15 +145,15 @@ class ParMapTest : StringSpec({
...
     checkAll(Arb.string()) { e ->
       (0 until 100).parMapOrAccumulate { _ ->
         raise(e)
       } shouldBe NonEmptyList(e, (1 until 100).map { e }).left()
     }
   }
 
-  "parMapNotNull is stack-safe" {
+  @Test fun parMapNotNullIsStackSafe() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:156]:
```suggestion
  @Test fun parMapNotNullIsStackSafe() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 5]
```diff
@@ -182,7 +183,7 @@ class ParMapTest : StringSpec({
     ).parMapNotNull { it.invoke() }
   }
 
-  "parMapNotNull results in the correct error" {
+  @Test fun parMapNotNullResultsInTheCorrectError() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:186]:
```suggestion
  @Test fun parMapNotNullResultsInTheCorrectError() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 6]
```diff
@@ -207,20 +208,20 @@ class ParMapTest : StringSpec({
...
 
-  "parMapNotNull discards nulls" {
+  @Test fun parMapNotNullDiscardsNulls() = runTest {
     (0 until 100).parMapNotNull { _ ->
       null
     } shouldBe emptyList()
   }
 
-  "parMapNotNull retains non-nulls" {
+  @Test fun parMapNotNullRetainsNonNulls() = runTest {
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:217]:
```suggestion
  @Test fun parMapNotNullRetainsNonNulls() = runTest(timeout = 60.seconds) {
```

File: arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt [hunk 7]
```diff
@@ -17,10 +16,12 @@ import io.kotest.property.arbitrary.string
 import io.kotest.property.checkAll
 import kotlin.time.Duration.Companion.milliseconds
 import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.test.runTest
 import kotlinx.coroutines.withTimeoutOrNull
+import kotlin.test.Test
```
Comment (nomisRev) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/ParMapTest.kt:21]:
```suggestion
import kotlin.test.Test
import kotlin.time.Duration.Companion.seconds
```

**Review by nomisRev — APPROVED** [review]
Hey @asmadsen,
Thank you for the contribution! 🙌 And congrats on your first contribution in Arrow 🥳

The timeout of kotlin-test is only 10s, and some of these tests are running for much longer on JS doing some stress testing. Solving out of the scope for this ticket, I added suggestions for increasing the timeout of said tests.

**Comment by asmadsen** [comment]
@nomisRev It seems like the timeout error in the JS tests is caused by mocha not using the timeout passed to the `runTest` function.

**Comment by serras** [comment]
@asmadsen sorry for taking over this PR with lots of changes, we've been investigating some weird interaction of `runTest` with `Atomic`

**Comment by asmadsen** [comment]
> @asmadsen sorry for taking over this PR with lots of changes, we've been investigating some weird interaction of `runTest` with `Atomic`

No problem. I have tried to run the tests locally however I struggle especially with `tvosSimulatorArm64Test`, it results in `java.lang.IllegalStateException: command '/usr/bin/xcrun' exited with errors (exit code: 137)`. I have made sure to install the latest XCode and all simulators.

Is it at all possible to run the tests locally, if so is there any guide or list of prerequisites for getting it working?

**Pull request merged**